tapgarden/planter.go

```diff
@@ -1787,9 +1895,50 @@ func (c *ChainPlanter) sealBatch(ctx context.Context, params SealParams,
 			return nil, fmt.Errorf("witness has no matching "+
 				"seedling: %v", wit)
 		}
+
 		externalWitnesses[wit.GenID] = wit
 	}
 
+	// Extract witnesses from signed group virtual PSBTs.
+	for idxPsbt := range params.SignedGroupVirtualPsbts {
+		psbtPacket := params.SignedGroupVirtualPsbts[idxPsbt]
+
+		groupReqMatch := matchPsbtToGroupReq(
+			psbtPacket, groupReqs,
+		).UnwrapToPtr()
+
+		// Ensure that a matching group key reveal has been found.
+		if groupReqMatch == nil {
+			return nil, fmt.Errorf("failed to find matching " +
+				"group key request for signed group virtual " +
+				"PSBT")
+		}
+
+		// Ensure that an external witness has not already been
+		// specified for the given genesis asset ID.
+		genesisAssetID := groupReqMatch.NewAsset.ID()
+
+		if _, ok := externalWitnesses[genesisAssetID]; ok {
+			return nil, fmt.Errorf("signed PSBT is a duplicate "+
+				"witness for asset ID: %v", genesisAssetID)
+		}
+
+		// Extract the witness stack from the signed PSBT.
+		witnessStack, err := DeserializeWitnessStack(
+			psbtPacket.Inputs[0].FinalScriptWitness,
```

Comment:
Alternatively, we should be able to finalize the PSBT at this point. Then we can just read the `Witness` attribute as normal.

Reply:
Yes, I think we should do that in any case, to make sure the TX is actually finalizable, which adds some additional validation.
Also, Ledger puts the signature in a different field (`Inputs[0].TaprootKeySpendSig`), which the following two calls take care of:
```
		psbt.MaybeFinalizeAll(packet)
		psbt.Extract(packet)
```